watertap/unit_models/ion_exchange_0D.py

```diff
@@ -141,85 +246,103 @@ class IonExchangeODData(InitializationMixin, UnitModelBlockData):
         ),
     )
 
+    CONFIG.declare(
+        "isotherm",
+        ConfigValue(
+            default=IsothermType.langmuir,
+            domain=In(IsothermType),
+            description="Designates the isotherm type to use for equilibrium calculations",
+        ),
+    )
+
     def build(self):
         super().build()
 
-        ### REFERENCES ###
-        # LeVan, M. D., Carta, G., & Yon, C. M. (2019).
-        # Section 16: Adsorption and Ion Exchange.
-        # Perry's Chemical Engineers' Handbook, 9th Edition.
-
-        # Crittenden, J. C., Trussell, R. R., Hand, D. W., Howe, K. J., & Tchobanoglous, G. (2012).
-        # Chapter 16: Ion Exchange.
-        # MWH's Water Treatment (pp. 1263-1334): John Wiley & Sons, Inc.
-
-        # DOWEX Ion Exchange Resins Water Conditioning Manual
-        # https://www.lenntech.com/Data-sheets/Dowex-Ion-Exchange-Resins-Water-Conditioning-Manual-L.pdf
-
-        # Inamuddin, & Luqman, M. (2012).
-        # Ion Exchange Technology I: Theory and Materials.
-
-        # Michaud, C.F. (2013)
-        # Hydrodynamic Design, Part 8: Flow Through Ion Exchange Beds
-        # Water Conditioning & Purification Magazine (WC&P)
-        # https://wcponline.com/2013/08/06/hydrodynamic-design-part-8-flow-ion-exchange-beds/
-
-        ion_set = self.config.property_package.ion_set
+        comps = self.config.property_package.component_list
         target_ion = self.config.target_ion
+
         self.target_ion_set = Set(
             initialize=[target_ion]
         )  # create set for future development of multi-component model
+        inerts = comps - self.target_ion_set
 
-        if "+" in target_ion:
+        if len(self.target_ion_set) > 1:
```

Comment:
So I meant to take this conditional out because I don't think it is possible for ``len(self.target_ion_set) > 1``. The only component that gets put in ``target_ion_set`` is ``target_ion`` which is set via CONFIG. Since the domain is ``str``, it seems anything that you pass as ``target_ion`` gets coerced to a ``str`` regardless. So if you pass ``["Ca_2+". "Na_+"]``, ``target_ion_set`` ends up with a single member ``'["Ca_2+". "Na_+"]'``, which then triggers a ``KeyError`` on the line 273 because ``'["Ca_2+". "Na_+"]'`` is ``not valid for indexed component fs.properties.charge_comp``. 

So I don't think this conditional is doing anything, but I am open to suggestions for how to implement the check needed. I see in GAC you have the domain as a list for ``target_species``... maybe that is the way to go?

Reply:
Maybe a list of strings as the expectation?